### PR TITLE
#25044 fix to pp archved multi language content

### LINKED ISSUE #24705: PP a folder with Archived Content Fails
### Parent Issue

_No response_

### Problem Statement

If you try to PP folder that contains archived content, if fails.

`Failed to publish because an error occurred: An error occurred when processing Workflow Task in '/data/shared/assets/bundles/01GYFRD81FZ8G699P9TDQ2H3MD/working/www.loans.com.au/1/422974622f905bcab3338eadf52e114e.contentworkflow.xml' with Title: webb-Nebula.png And ID '8ed7f3b2-db83-4adf-b447-826c4c021cc0': ERROR: insert or update on table "workflow_task" violates foreign key constraint "fk_workflow_task_asset" Detail: Key (webasset)=(422974622f905bcab3338eadf52e114e) is not present in table "identifier".{ "SQL": ["INSERT INTO workflow_task (id, creation_date, mod_date, due_date, created_by, assigned_to, belongs_to, title, description, status, webasset, language_id) values (?,?,?,?,?,?,?,?,?,?,?,?)"], "maxRows": [-1], "offest": [0], "params": [ "8ed7f3b2-db83-4adf-b447-826c4c021cc0", "2023-04-21 02:36:25.867", "2023-04-21 02:36:32.992", null, "7a2c45a2-4496-488b-a16a-58af39c716b8", "7a2c45a2-4496-488b-a16a-58af39c716b8", null, "webb-Nebula.png", "", "d6b095b6-b65f-4bdb-bbfd-701d663dfee2", "422974622f905bcab3338eadf52e114e", 1 ] }`

### Steps to Reproduce

1- Configure PP.
2- Create a folder in the sender.
3- Upload 2 files to the created folder.
4- Archive one of the files.
5- PP the whole folder.
6- Get the error.

### Acceptance Criteria

We should be able to PP folder with files whichever state they are.

### dotCMS Version

23.01.2 LTS

### Proposed Objective

Customer Support

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/111095

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

As a workaround since the content is already archived probably won't be needed anymore, so delete it and try again to PP the folder.

### Sub-Tasks & Estimates

_No response_

### LINKED ISSUE #25044: PP content set as working state instead of live state when there is archived version in other language
### Parent Issue

_No response_

### Problem Statement

When a live content is push published and there is an archived version of the content in other language, the content is push published to working state instead of live state as expected.

### Steps to Reproduce

1. Create a new content in English (or default language)
2. Publish the content
3. Create a version of the content in Spanish or another language
4. Archive only the Spanish version (the English version is still live)
5. Push publish the content
6. Go the receiver: the English version was push published as working instead of live

### Acceptance Criteria
Content should be push published in the same state in the receiver.

### dotCMS Version

22.03.6 and  23.05

### Proposed Objective

Customer Support

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
https://dotcms.zendesk.com/agent/tickets/111625
https://dotcms.zendesk.com/agent/tickets/111552

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by dsolistorres**
### <samp>🤖 Generated by Copilot at 2b0b7e3</samp>

*  Add a condition to check if the content exists in the receiver before archiving it in `handleContents` method in `ContentHandler` class, and save it first if not, to avoid losing the content data ([link](https://github.com/dotCMS/core/pull/25180/files?diff=unified&w=0#diff-39e5f35ab504d57b9244a02e68aae5395e59ea83fa525875137f76fb967279beL334-R358))
*  Add the code that checks if the content has been pushed as archived to the receiver using the `pushedIdsToIgnore` set of  content id/language pairs in `handleContents` method in `ContentHandler` class, and skip the content if true ([link](https://github.com/dotCMS/core/pull/25180/files?diff=unified&w=0#diff-39e5f35ab504d57b9244a02e68aae5395e59ea83fa525875137f76fb967279beR401-R410))
*  Modify the logic of the `unpublish` method in `ContentHandler` class to check if the content is live before unpublishing it, and to find the existing content version info based on the `existingContentMap` map if there are unique fields that cause the pushed content being mapped to an existing content with a different identifier ([link](https://github.com/dotCMS/core/pull/25180/files?diff=unified&w=0#diff-39e5f35ab504d57b9244a02e68aae5395e59ea83fa525875137f76fb967279beL1136-R1201))

### Checklist
- [x] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
Using a content identifier/language pair for the `pushedIdsToIgnore` set, fixes the problem when there are multiple language versions for the content being pushed, and one of the versions is archived. Before the fix, the publishing of live versions was skipped by the `handleContents` method.

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java [hunk 1]
```diff
@@ -331,14 +333,29 @@ private void handleContents(final Collection<File> contents, final File folderOu
 						// This operation (PUBLISH) publishes/un-publishes a Content that is NOT archived. The un-publishing
 						// is taken care of at the end of this handleContents() method
 						publish(content, folderOut, systemUser, wrapper, isHost,remoteLocalLanguages);
+						Logger.debug(this, "Content saved: " + content.getIdentifier()
```
Comment (jdotcms) [inline dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java:336]: you should use Logger.debug(class, Supplier) such sa

Logger.debug(this, ()-> "Content saved: " + ...)

This makes the lazy creation of the msg so you safe mem if the debug is not activate

File: dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java [hunk 2]
```diff
@@ -331,14 +333,29 @@ private void handleContents(final Collection<File> contents, final File folderOu
...
+								+ " , inode: " + content.getInode() + " , language: " + content.getLanguageId());
 						// Store the IDs that will be archived in the receiver
-						pushedIdsToIgnore.add(content.getIdentifier());
+						final Pair<String,Long> idAndLanguageKey = Pair.of(content.getIdentifier(),
+								remoteLocalLanguages.getRight());
+						pushedIdsToIgnore.add(idAndLanguageKey);
 					} else {
 						// Finally, this operation (UNPUBLISH) deletes a Content altogether
 						unpublish(content, systemUser, isHost, remoteLocalLanguages, false);
+						Logger.debug(this, "Content deleted: " + content.getIdentifier()
```
Comment (jdotcms) [inline dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java:357]: Content deleted? shouldn't be unpublish?

File: dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java [hunk 3]
```diff
@@ -331,14 +333,29 @@ private void handleContents(final Collection<File> contents, final File folderOu
...
 						// This operation (PUBLISH) flags the pushed Content as Archived, WITHOUT deleting it
+						boolean localContentAlreadyExists = APILocator.getContentletAPI().isContentlet(content.getInode());
+						if (!localContentAlreadyExists) {
+							// If the content doesn't exist in the local instance yet, we need to save it first
+							publish(content, folderOut, systemUser, wrapper, isHost, remoteLocalLanguages);
+							Logger.debug(this, "Content saved before archive: " + content.getIdentifier()
+									+ " , inode: " + content.getInode() + " , language: " + content.getLanguageId());
+						}
 						unpublish(content, systemUser, isHost, remoteLocalLanguages, true);
+						Logger.debug(this, "Content archived: " + content.getIdentifier()
```
Comment (jdotcms) [inline dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java:348]: Content deleted? shouldn't be unpublish?

Comment (dsolistorres) [reply dotCMS/src/enterprise/java/com/dotcms/enterprise/publishing/remote/handler/ContentHandler.java]: I renamed the method to `archiveOrDelete` for clarity.

**Pull request merged**